node/src/actors/rad_manager/actor.rs

```diff
@@ -10,6 +10,11 @@ impl Actor for RadManager {
     fn started(&mut self, _ctx: &mut Self::Context) {
         log::debug!("RadManager actor has been started!");
     }
+
+    fn stopped(&mut self, _ctx: &mut Self::Context) {
+        // RadManager is expected to restart on panic
+        //System::current().stop_with_code(1);
```

Comment:
In this case, when the RadManager dies, it looks that it would never recover

Reply:
I assumed that it is restarted automatically, will try to add tests to see which actors restart and which ones stop the system when they panic.

Reply:
Currently all actors stop the system on panic, (except `Session` and `JsonRpc`, which handle client connections and disconnect on panic). `RadManager` executes data requests inside a future, and that makes it immune to radon panics. It's not that the `RadManager` restarts automatically, it's that it never stops.